jenkinsfile-examples/msbuild/Jenkinsfile

```
@@ -4,8 +4,11 @@ node {
 
 	stage 'Build'
 		bat 'nuget restore SolutionName.sln'
-		bat "\"${tool 'MSBuild'}\" SolutionName.sln /p:Configuration=Release /p:Platform=\"Any CPU\" /p:ProductVersion=1.0.0.${env.BUILD_NUMBER}"
-
+		bat """
+			setlocal enabledelayedexpansion
+			\"${tool 'MSBuild'}\" SolutionName.sln /p:Configuration=Release /p:Platform=\"Any CPU\" /p:ProductVersion=1.0.0.!BUILD_NUMBER!
+			endlocal
+		"""
```

Perhaps something like
```suggestion
		def msbuild = tool 'MSBuild'
		bat msbuild + " SolutionName.sln /p:Configuration=Release /p:Platform="Any CPU" /p:ProductVersion=1.0.0.%BUILD_NUMBER%"
```

Granted, this theoretically still has the problem described in https://plugins.jenkins.io/safe-batch-environment-filter/ but we know that `%BUILD_NUMBER%` is numeric. Perhaps mention it and move on?